lib/pbench/server/__init__.py

```diff
@@ -125,6 +124,27 @@ def mocked_time():
             + " ".join([f"WONT-INDEX.{i:d}" for i in range(1, 12)])
         )
 
+    def _get_conf(self, section, option):
+        """
+        _get_conf - get the option from the section, raising
+        ValueError if it is empty, return the option value as a string.
+        """
+        option_val = self.conf.get(section, option)
+        if not option_val:
+            raise ValueError(f"option {option} in section {section} is empty")
+        return option_val
+
+    def _get_valid_dir_option(self, req_val, section, option):
+        """_get_valid_dir_option - get the directory option
+        from the given section, raising a ValueError if it
+        is not a directory, returning a Path directory object.
+        """
+        dir_val = self._get_conf(section, option)
+        dir_path = Path(dir_val)
+        if not dir_path.is_dir():
+            raise BadConfig(f"Bad {req_val}={dir_path}")
+        return dir_path
+
```

Comment:
Add an externally callable method to do parts of what _get_valid_dir_option() does.